examples/python-cicd-with-cloudbuilder/README.md

```diff
@@ -0,0 +1,135 @@
+Name                     Stmts   Miss  Cover
+--------------------------------------------
+my_module/__init__.py        1      0   100%
+my_module/my_module.py       4      0   100%
+--------------------------------------------
+TOTAL                        5      0   100%
+
+
+```
+
+Now that our tests are working locally, we want to configure GCP to run them every time we push new code.
+
+## 3. Building a Python Cloud Builder
+In order to execute python based tests in cloud builder you need a python cloud builder. 
+A cloud builder is just a docker container with the stuff you need in it. Google doesn't distribute a 
+python cloud builder, so we have to build one ourselves. I've created one in '/python-cloud-builder.' All you need to do 
+is run this:
+
+```
+cd python-cloud-builder
+gcloud builds submit --config=cloudbuild.yaml .
+
+```
+
+This creates a custom cloud builder in your container registry called gcr.io/$PROJECT_ID/python-cloudbuild that you can
+use to run tests with.
+
+## 4. Create a Cloud Build Trigger
+* On the GCP console navigate to 'Cloud Build' > 'Triggers'
+* Add a trigger
+* Choose cloud source repository
+* Pick the repo you created in step 1.
+* Assuming you want the trigger to fire on any branch, accept the default trigger type and regex
+* Choose 'cloud build configuration file' from 'build configuration'
+* Click create trigger
+
+
+## 5. Create a cloudbuild.yaml file that executes your tests and runs pylint
+Create a file called cloudbuild.yaml in the root of your project and add the steps needed to execute your unit tests, 
+using the cloud builder we created in step 3. 
+
+*Note: Each cloudbuilder step is a separate, ephemerial run of a docker container, however the /workspace/ directory is
+preserved between runs. One way to carry python packages forward is to use a virtualenv housed in /workspace/*
+
+
+```
+steps:
+- name: 'gcr.io/$PROJECT_ID/python-cloudbuild'
+  entrypoint: '/bin/bash'
+  args: ['-c','virtualenv /workspace/venv' ]
+- name: 'gcr.io/$PROJECT_ID/python-cloudbuild'
+  entrypoint: 'venv/bin/pip'
+  args: ['install', '-V', '-r', 'requirements.txt']
+- name: 'gcr.io/$PROJECT_ID/python-cloudbuild'
+  entrypoint: 'venv/bin/python'
+  args: ['-m', 'pytest', '-rA']
+- name: 'gcr.io/$PROJECT_ID/python-cloudbuild'
+  entrypoint: 'venv/bin/pylint'
+  args: ['my_module/']
+```
+
+## Wrap Up
+That's all there is to it! 
+
+From here you can inspect your builds in Cloud Builder's history.  You can also build in third-party integrations via
+PubSub. There's some documentation on how [here.](https://cloud.google.com/cloud-build/docs/configure-third-party-notifications)
+
+If we wanted to automatically deploy this code we could add another step that enabled continuous delivery too. Prebuilt cloud builders
+exist for gcloud, kubectl, etc. The full list can be found at [https://github.com/GoogleCloudPlatform/cloud-builders.](https://github.com/GoogleCloudPlatform/cloud-builders)
```

Comment:
Nit: A pet peeve of mine is when parts of a sentence are hyperlinked and the period at the end is included as well. However, I accept this is my own insanity so I won't hold you to this. Feel free to ignore.

Reply:
I don't have a strong feeling either way about this, so I did it your way